build/build_system.bzl

```diff
@@ -235,3 +246,101 @@ def git_or_local_repository(name, branch, **kwargs):
             branch = branch,
             **kwargs
         )
+
+def _kong_install_impl(ctx):
+    outputs = []
+    strip_path = ctx.attr.strip_path
+    label_path = ctx.attr.src.label.workspace_name + "/" + ctx.attr.src.label.name
+    if not strip_path:
+        strip_path = label_path
+    prefix = ctx.attr.prefix
+    if prefix:
+        prefix = prefix + "/"
+
+    for file in ctx.files.src:
+        # skip top level directory
+        if file.short_path.endswith(label_path) or file.short_path.endswith(strip_path):
+            continue
+
+        # strip ../ from the path
+        path = "/".join(file.short_path.split("/")[1:])
+
+        # skip foreign_cc generated copy_* targets
+        if path.startswith(ctx.attr.src.label.workspace_name + "/copy_"):
+            continue
+
+        target_path = path.replace(strip_path + "/", "")
+        full_path = "%s/%s%s" % (KONG_VAR["BUILD_NAME"], prefix, target_path)
+
+        # use a fake output, if we are writing a directory that may collide with others
+        # nop_path = "%s-nop-farms/%s/%s/%s" % (KONG_VAR["BUILD_NAME"], strip_path, prefix, target_path)
+        # output = ctx.actions.declare_file(nop_path)
+        # ctx.actions.run_shell(
+        #     outputs = [output],
+        #     inputs = [file],
+        #     command = "(mkdir -p {t} && chmod -R +rw {t} && cp -r {s} {t}) >{f}".format(
+        #         t = full_path,
+        #         s = file.path,
+        #         f = output.path,
+        #     ),
+        # )
+        output = ctx.actions.declare_file(full_path)
+        ctx.actions.run_shell(
+            outputs = [output],
+            inputs = [file],
+            command = "cp -r %s %s" % (file.path, output.path),
+        )
+
+        outputs.append(output)
+
+        if full_path.find(".so.") >= 0 and ctx.attr.create_dynamic_library_symlink:
+            el = full_path.split(".")
+            si = el.index("so")
+            sym_paths = []
+            if len(el) > si + 2:  # has more than one part after .so like libX.so.2.3.4
```

Comment:
``` suggestion
            if len(el) > si + 2:  # has more than one part after .so like libX.so.2.3.4 / libluajit.so-2.1.rolling
````

https://www.freelists.org/post/luajit/LuaJIT-uses-rolling-releases

Reply:
we no longer uses the `ROLLING` suffix in produced library in this PR, is it still needed?

Reply:
```
$ ll bazel-bin/build/kong-dev/openresty/luajit/lib
total 3384
drwxrwxrwx 2 root root    4096 Apr 22 04:04 ./
drwxrwxrwx 6 root root    4096 Apr 22 03:53 ../
lrwxrwxrwx 1 root root      28 Apr 22 04:04 libluajit-5.1.so -> libluajit-5.1.so.2.1.ROLLING*
lrwxrwxrwx 1 root root      28 Apr 22 04:04 libluajit-5.1.so.2 -> libluajit-5.1.so.2.1.ROLLING*
-rwxrwxrwx 1 root root 3453624 Apr 22 03:53 libluajit-5.1.so.2.1.ROLLING*
```

This is what the venv generated, not sure if any affects.

Reply:
I feel like this not a common/standard way to name so libraries, it must ends with `.so` or a digit (https://stackoverflow.com/questions/663209/can-someone-explain-linux-library-naming). So let's not keep the `.ROLLING` suffix here I propose.